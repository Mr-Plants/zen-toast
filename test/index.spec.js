import {Toast} from "../index";
import Vue from 'vue';

describe('zen-toast_v_1.0.0', () => {
  describe('dom', () => {
    test('toast basic dom', async () => {
      const toast = Toast('爱过～');
      expect(toast.$el.outerHTML).toMatchSnapshot();
      expect(document.querySelector('.toast-overlay')).toBeTruthy();

      expect(document.querySelector('.base')).toBeTruthy();
      expect(toast.timer).toBeTruthy();

      // 1.5s后自动消失
      await later(1500);
      expect(toast.timer).toBeFalsy();
      expect(document.querySelector('.toast-overlay')).toBeFalsy();
    })

    test('toast success dom', async () => {
      const toast = Toast.success('爱过～');
      expect(toast.$el.outerHTML).toMatchSnapshot();
      expect(document.querySelector('.success')).toBeTruthy();
      expect(document.querySelector('.icon')).toBeTruthy();
    })

    test('toast fail dom', async () => {
      const toast = Toast.fail('爱过～');
      expect(toast.$el.outerHTML).toMatchSnapshot();
      expect(document.querySelector('.fail')).toBeTruthy();
      expect(document.querySelector('.icon')).toBeTruthy();
    })

    test('toast loading dom', async () => {
      const toast = Toast.loading('loading...');
      expect(toast.$el.outerHTML).toMatchSnapshot();
      expect(document.querySelector('.loading')).toBeTruthy();
      expect(document.querySelector('.icon')).toBeTruthy();
    })
  })

  describe('props', () => {
    test('props should set correct', async () => {
      const toast = Toast({
        message: 'message'
      })
      expect(toast.$el.outerHTML).toMatchSnapshot();

      await later();
      expect(toast.message).toBe('message');
      expect(toast.duration).toBe(1.5);
      expect(toast.closeOnClick).toBeFalsy()
      expect(toast.type).toBe('base');
    })

    test('message length should less than 18', async () => {
      const toast = Toast({
        message: 'messagemessagemessagemessagemessagemessagemessagemessage'
      })
      expect(toast.$el.outerHTML).toMatchSnapshot();

      await later();
      expect(toast.message).toHaveLength(18);
      expect(toast.duration).toBe(3);
    })
  })

  describe('event', () => {
    test('click overlay should close toast', () => {
      const toast = Toast({
        duration: 0,
        message: 'message',
        closeOnClick: true
      });
      expect(toast.$el.outerHTML).toMatchSnapshot();
      toast.$el.click();
      expect(document.querySelector('.toast-overlay')).toBeFalsy();
    })
  })

  describe('plugin', () => {
    test('install method should invoke', () => {
      Vue.use(Toast);
      expect(Vue.prototype.$toast).toEqual(Toast);
    })
  })

});

function later(delay = 0) {
  return new Promise(resolve => {
    setTimeout(resolve, delay);
  });
}
