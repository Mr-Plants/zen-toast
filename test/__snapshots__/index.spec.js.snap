// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zen-toast_v_1.0.0 dom toast basic dom 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content base\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">爱过～</span></div></div>"`;

exports[`zen-toast_v_1.0.0 dom toast fail dom 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content fail\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">爱过～</span></div></div>"`;

exports[`zen-toast_v_1.0.0 dom toast loading dom 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content loading\\"><span class=\\"icon\\"><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i><i></i></span> <span class=\\"message\\">loading...</span></div></div>"`;

exports[`zen-toast_v_1.0.0 dom toast success dom 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content success\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">爱过～</span></div></div>"`;

exports[`zen-toast_v_1.0.0 event click overlay should close toast 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content base\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">message</span></div></div>"`;

exports[`zen-toast_v_1.0.0 props message length should less than 18 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content base\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">messagemessagemessagemessagemessagemessagemessagemessage</span></div></div>"`;

exports[`zen-toast_v_1.0.0 props props should set correct 1`] = `"<div class=\\"toast-overlay\\"><div class=\\"content base\\"><span class=\\"icon\\"><!----></span> <span class=\\"message\\">message</span></div></div>"`;
